### DISCUSSION TITLE
Lesson 7: Everything works out except exception was thrown try on Etherscan

### DISCUSSION THREAD

**Discussion by LBCrypto** [smartcontractkit/full-blockchain-solidity-course-py#151] [Q&A] [answered]
After I ran the deploy script everything went ok. But when I tried to use the contract on rinkeby etherscan the endlottery function threw an exception, metamask says: Transaction Error. Exception thrown in contract code.
If I make tx anyway, the transaction will fail. I would appreciate it if anyone know the solution for this. BTW I cloned this git code, also have the same issue.

Update: I tried to use python do the lottery 2 times and it works. Then I tried to use etherscan the same problem occurs. But then I used python to end the lottery it works again. Anyone see the same problem?

**Comment by LBCrypto** [comment, marked as answer]
I think I have found the reason. The contract itself lack the ability to fund the link. Fund link function is called by python helpful script.

**Reply by PatrickAlphaC** [reply]
boom!
